agent/background.go

```diff
@@ -23,10 +23,14 @@ func (a *Agent) startHeartbeat(ctx context.Context, cancel context.CancelFunc, t
 	ticker := time.NewTicker(heartbeatInterval)
 	defer ticker.Stop()
 
+	var hasSentHeartbeat bool
 	for {
 		select {
 		case <-ticker.C:
 			signalBeat, err = a.doHeartbeat(ctx, tc)
+			if hasSentHeartbeat {
```

Comment:
We should move this [block](https://github.com/evergreen-ci/evergreen/pull/6771/files#diff-569d49a4c046eb603023b873eea7110382401df292e83230a95862469c2029f9R46-R56) before we continue in case the err is not nil right?